internal/charmstore/store.go

```diff
@@ -1047,3 +1015,17 @@ func (s *Store) SynchroniseElasticsearch() error {
 	}
 	return nil
 }
+
+// EntityResolvedURL returns the ResolvedURL for the entity.
+// It requires the PromulgatedURL field to have been
+// filled out in the entity.
+func EntityResolvedURL(e *mongodoc.Entity) *router.ResolvedURL {
```

Comment:
I know this is indirectly tested everywhere, but it would be nice to have unit tests for the function.